src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect } from "react";
+
+export const Modal = ({
+  name,
+  onClose,
+  isOpen,
+  children,
+  contentClass = "",
+}) => {
+  useEffect(() => {
+    const handleEscape = (e) => {
+      if (e.key === "Escape") {
+        onClose();
+      }
+    };
+    if (isOpen) {
+      document.addEventListener("keydown", handleEscape);
+    }
+    return () => {
+      document.removeEventListener("keydown", handleEscape);
+    };
+  }, [isOpen, onClose]);
+
+  const handleOverlay = (e) => {
+    if (e.target === e.currentTarget) {
+      onClose();
+    }
+  };
+
+  return isOpen ? (
+    <div
+      className={`modal modal_type_${name} ${isOpen ? "modal__opened" : ""}`}
```

Comment:
`modal__opened` is an element of the `modal` block, but according to the task and BEM you need to use a **modifier** with only **one** low dash: `modal_opened`